### PR TITLE
fix #250: add runtime inputs (js, json) to output sources

### LINKED ISSUE #250: [Bug]: JSON data from `ts_project` not found at runtime
### What happened?

After discussion in the Bazel Slack in the `#javascript` channel, opening an issue here to track this apparent bug.

Given a a `ts_project` that resolves data as a "JSON module", when running a `js_binary` that depends on that `ts_project`, the JSON file is not found at runtime. The expected behavior is that the data would be included at runtime. It is also expected that the data be available at compile time since it may imported, not only by it's local package target, but by other `ts_project`s that depend on it.

### Version

Development (host) and target OS/architectures:

Output of `bazel --version`: `bazel 5.3.2`

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:  `rules_ts-1.0.0-rc6`

Language(s) and/or frameworks involved: TypesScript, JavaScript

### How to reproduce

A branch reproducing the issue can be found here: https://github.com/loudmouth/bazel-rules_js-poc/tree/ts_project-data

Just try to running `bazel run //app/app-a:main` and the output will be: 

"Error: Cannot find module './data.json'"

```shell
├── BUILD.bazel
├── WORKSPACE
├── app
│   └── app-a
│       ├── BUILD.bazel
│       ├── index.ts
├── lib
│   ├── lib-a
│   │   ├── BUILD.bazel
│   │   ├── index.ts
│   ├── lib-data
│   │   ├── BUILD.bazel
│   │   ├── data.json
│   │   ├── index.ts
```

### Any other information?

Linking to the [Slack thread](https://bazelbuild.slack.com/archives/CEZUUKQ6P/p1668518442533219) since there was already discussion as to (a) possible fix(es)

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
I have split this into two commits for ease of readability. LMK if you'd like me to squash or open two PRs.

File: ts/private/ts_config.bzl [hunk 1]
```diff
@@ -118,7 +118,7 @@ def _filter_input_files(files, allow_js, resolve_json_module):
         f
         for f in files
         # include typescript, json & declaration sources
-        if _lib.is_ts_src(f.basename, allow_js) or _lib.is_json_src(f.basename, resolve_json_module) or _lib.is_typings_src(f.basename)
+        if _lib.is_ts_src(f.basename, allow_js, resolve_json_module) or _lib.is_typings_src(f.basename)
```
Comment (gzm0) [inline ts/private/ts_config.bzl:121]: /cc @gregmagolan

**Comment by gzm0** [comment]
I have realized that after this change, js and json input files are treated equally (modulo the flag that allows them). So I have added an additional commit that formalizes this (i.e. removes is_json_src).

LMK what you think. Cleanest might be to split the last commit over the first two.

File: examples/jsx/expected_outputs.txt [hunk 2]
```diff
@@ -2,6 +2,7 @@ examples/jsx/out/a.d.ts
 examples/jsx/out/a.d.ts.map
 examples/jsx/out/b.d.ts
 examples/jsx/out/b.d.ts.map
+examples/jsx/b.jsx
```
[the remark this answers is not in the document]
Comment (thesayyn) [reply examples/jsx/expected_outputs.txt:5]: this doesn't look right to me. https://github.com/aspect-build/rules_ts/blob/5ea2d011369bf371d2431871bd178d6893766392/examples/jsx/BUILD.bazel#L14 there's an outdir attribute so `b.jsx` should be `out/b.js.x`.

[the remark this answers is not in the document]
Comment (gzm0) [reply examples/jsx/expected_outputs.txt:5]: Ah, ah indeed. This is not right. I missed the outdir.

IIUC this means:
- I need to go back to the drawing board on this PR (I might submit just the refactoring).
- Setting `out_dir` might be a viable workaround.

**Comment by gregmagolan** [comment]
Thanks for the PR @gzm0 . Myself or @alexeagle needs to load up the context on this one again and make sure this doesn't regress any current behaviour that users are depending on.

**Comment by gzm0** [comment]
Thanks for the update. Please LMK if there is anything I can help (e.g. adding additional test cases that are relevant).

**Review by alexeagle — APPROVED** [review]
I studied this for a bit, and I think it's correct. I hope our test coverage is good in case I missed something.
Thanks! Sorry this took so long to get to.

**Comment by gzm0** [comment]
Rebased onto main.

**Comment by gzm0** [comment]
Gentle ping on this. Is there anything you need from my side for this? (or is this waiting for 2.0?).

**Pull request merged**